File: to_json.py
import os, json

path_to_forum_txt = "data/???.txt"

with open(path_to_forum_txt, 'r') as f:
    forum : list[str] = f.read().split("*-end-content-*")


forum_cleaned : list[str] = []

for msg in forum:
    lines : list[str] = msg.splitlines()
    content_start = False
    msg_clean : str = ""

    for line in lines:
        if line == "*-delimit-msg-*":
            line = ""
        elif line == "*-start-content-*":
            content_start = True

        if content_start:
            msg_clean += line + "\n"
        elif len(line) != 0:
            msg_clean += line + "\n"

    
    if len(msg_clean) != 0:
        forum_cleaned.append(msg_clean)


forum_cet_json = []

for msg in forum_cleaned:
    lines = msg.splitlines()
    d = dict()

    d["topic"] = lines[0][1:len(lines[0])-1].strip()
    d["date"] = lines[1][1:11].strip()
    d["hour"] = lines[1][12:20].strip()
    d["author"] = lines[1][34:].split("-")[0].strip()
    d["content"] = "\n".join(lines[3:]).strip()

    forum_cet_json.append(d)

with open("???.json", 'w', encoding="utf8") as f:
    json.dump(forum_cet_json, f, ensure_ascii=False)

File: to-json.js
const fs = require('node:fs')
const output = []
fs.readFile('./data/forum-cet.txt', 'utf8', (err, data) => {
  if (err) {
    console.error(err)
    return
  }
  const messages = data.split("*-delimit-msg-*")
  for (const message of messages) {
    const contents = message.split("*-start-content-*")
    if (contents.length == 1) {
      continue
    }
    
    const header = contents[0].split("\n")
    const topic = header[1].substring(1, header[1].length - 1).trim()
    const metas = header[2].substring(1, header[2].length-1).split("\t")
    const date = metas[0].split(" ")[0].trim()
    const hour = metas[0].split(" ")[1].trim()
    const author = metas[1].split("-")[0].substring(13).trim()
    const content = contents[1].split("*-end-content-*")[0].trim()

    output.push({
      topic,
      date,
      hour,
      author,
      content
    })
  }

  fs.writeFileSync('./data/forum-cet.json', JSON.stringify(output), 'utf8');
})
